lib/msf/core/rpc/v10/client.rb

```diff
@@ -69,10 +69,10 @@ def call(meth, *args)
     res = @cli.send_recv(req)
     @cli.close
 
-    if res and [200, 401, 403, 500].include?(res.code)
+    if res && [200, 401, 403, 500].include?(res.code)
       resp = MessagePack.unpack(res.body)
 
-      if resp and resp.kind_of?(::Hash) and resp['error'] == true
+      if resp && resp.kind_of?(::Hash) && resp['error']
```

Comment:
I thought this was already reverted?

Also, puts with no arg looks like a bug to you?  It looks more correct to me since it's the default behavior of puts, just like it is with print status etc.  To me puts '' looks like you forgot to fill in the string

> On Feb 28, 2015, at 12:52, Tod Beardsley <notifications@github.com> wrote:
> 
> In lib/msf/core/rpc/v10/client.rb:
> 
> >        resp = MessagePack.unpack(res.body)
> >  
> > -      if resp and resp.kind_of?(::Hash) and resp['error'] == true
> > +      if resp && resp.kind_of?(::Hash) && resp['error']
> I think I figured out what @bturner-r7 is saying here. resp['error'] mustn't be merely truthy (non-NilClass and non-FalseClass), it must actually == true.
> 
> —
> Reply to this email directly or view it on GitHub.
>